omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/PatientIdentifierResource1_8.java

```diff
@@ -214,17 +214,29 @@ public PatientIdentifier getByUniqueId(String uniqueId) {
 	 */
 	@Override
 	public PatientIdentifier save(PatientIdentifier delegate) {
-		// make sure it has already been added to the patient
 		boolean needToAdd = true;
 		for (PatientIdentifier pi : delegate.getPatient().getActiveIdentifiers()) {
 			if (pi.equals(delegate)) {
 				needToAdd = false;
 				break;
 			}
 		}
-		if (needToAdd)
+		
+		if (needToAdd) {
 			delegate.getPatient().addIdentifier(delegate);
-		return service().savePatientIdentifier(delegate);
+		}
+		service().savePatientIdentifier(delegate);
+		//service().savePatient(delegate.getPatient());
```

Comment:
What is the use of the above commented out line?

Reply:
My bad, will remove